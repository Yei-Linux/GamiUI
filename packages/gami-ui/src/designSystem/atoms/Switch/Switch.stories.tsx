import { Meta } from '@storybook/react'
import {
  getTemplate,
  stylesControl,
} from '../../../core/helpers/storybook.helper'

import Switch from '.'

const Template = getTemplate(Switch)

export default {
  title: 'Atoms/Switch',
  component: Switch,
  args: {},
  argTypes: {
    ...stylesControl,
  },
} as Meta

export const BasicActiveSwitch = Template.bind({})
BasicActiveSwitch.args = {
  width: 'NORMAL',
  shadow: '',
}

export const BasicInactiveSwitch = Template.bind({})
BasicInactiveSwitch.args = {
  width: 'NORMAL',
  shadow: '',
  defaultChecked: false,
}
