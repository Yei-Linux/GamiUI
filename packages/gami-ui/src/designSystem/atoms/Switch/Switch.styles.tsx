import styled from '@emotion/styled'
import {
  BorderType,
  HeightType,
  ShadowType,
  WidthType,
} from 'core/domain/types'
import { borderRadius, spacing, theme } from 'styles/tokens'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'

export const Switch = styled.button<{
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
}>`
  position: relative;

  min-width: 44px;
  height: 22px;

  margin: ${spacing.margin.none};
  padding: 0px 3px;
  border: ${spacing.border.none};

  transition: all 0.1s;
  user-select: none;

  &:hover {
    cursor: pointer;
  }

  &.checked {
    background: ${theme.light.primary.jordyBlue};
  }

  &.notChecked {
    background: ${theme.light.neutral[400]};
  }

  ${({ $border, $shadow, $width, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'lg',
      shadow: $shadow || 'sm',
      width: $width,
      heigth: $heigth,
    })};
`

export const SwitchBall = styled.div`
  position: absolute;
  top: 8%;

  width: 18px;
  height: 18px;

  border-radius: ${borderRadius.md};

  background: ${theme.light.neutral[800]};

  transition: all 0.2s ease-in-out;

  &.checked {
    left: calc(100% - 20px);
  }

  &.notChecked {
    left: 5%;
  }
`
