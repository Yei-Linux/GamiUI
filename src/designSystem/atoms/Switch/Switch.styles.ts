import styled from "@emotion/styled";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const SwitchWrapper = styled.button<{
  isChecked?: boolean;
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  position: relative;

  margin: 0;
  padding: 0px 3px;
  min-width: 44px;
  height: 22px;
  border: 0;
  cursor: pointer;
  transition: all 0.2s;
  user-select: none;

  background: ${(props: any) =>
    props.isChecked ? colorLight.primary.one : colorLight.neutral.four};

  ${(props: any) => twinStyles(props)};
`;

export const SwitchBall = styled.div<{ isChecked?: boolean }>`
  position: absolute;
  top: 8%;
  left: ${(props: any) => (props.isChecked ? "calc(100% - 20px)" : "5%")};

  width: 18px;
  height: 18px;
  border-radius: 15px;
  transition: all 0.2s ease-in-out;
  background: white;
`;
