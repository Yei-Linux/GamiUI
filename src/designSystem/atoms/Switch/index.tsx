export { default } from "./Switch";
export { options } from "./constants";
