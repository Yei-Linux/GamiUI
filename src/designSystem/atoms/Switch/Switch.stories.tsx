import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Switch from ".";

const Template = getTemplate(Switch);

export default {
  title: "Atoms/Switch",
  component: Switch,
  args: { ...Switch.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const BasicActiveSwitch = Template.bind({});
BasicActiveSwitch.args = {
  width: "NORMAL",
  shadow: "",
};

export const BasicInactiveSwitch = Template.bind({});
BasicInactiveSwitch.args = {
  width: "NORMAL",
  shadow: "",
  defaultChecked: false,
};
