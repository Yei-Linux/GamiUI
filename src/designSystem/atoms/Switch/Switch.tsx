import React, { useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { SwitchBall, SwitchWrapper } from "./Switch.styles";

export interface ISwitch extends IGeneralProps {
  defaultChecked: boolean;
  /**
   * Function to detect changes
   */
  onChangeFormItem?: any;
}

const Switch = ({ onChangeFormItem, defaultChecked, ...args }: ISwitch) => {
  const [isChecked, setIsChecked] = useState(defaultChecked);

  const toggle = () => {
    onChangeFormItem(!isChecked);
    setIsChecked(!isChecked);
  };

  return (
    <SwitchWrapper isChecked={isChecked} {...args} onClick={() => toggle()}>
      <SwitchBall isChecked={isChecked} />
    </SwitchWrapper>
  );
};

Switch.defaultProps = {
  defaultChecked: true,
  border: "ROUNDED",
  width: "NORMAL",
};

export default Switch;
