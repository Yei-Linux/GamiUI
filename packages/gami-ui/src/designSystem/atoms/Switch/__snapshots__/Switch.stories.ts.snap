// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Switch Basic Active Switch 1`] = `
<button
  className="css-ssnb1t-SwitchWrapper egh0tog0"
  onClick={[Function]}
  width="NORMAL"
>
  <div
    className="css-fuaihs-SwitchBall egh0tog1"
  />
</button>
`;

exports[`Storyshots Atoms/Switch Basic Inactive Switch 1`] = `
<button
  className="css-13kjyr7-SwitchWrapper egh0tog0"
  onClick={[Function]}
  width="NORMAL"
>
  <div
    className="css-1lmkk7b-SwitchBall egh0tog1"
  />
</button>
`;
